plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -0,0 +1,173 @@
+    mutationToDom: function () {
+        const container = document.createElement('mutation');
+        container.setAttribute('inputs', this.inputCount_);
+        return container;
+    },
+
+    domToMutation: function (xmlElement) {
+        this.inputCount_ = parseInt(xmlElement.getAttribute('inputs'), 10);
+        this.updateShape_();
+    },
+
+    saveExtraState: function () {
+        return {
+            'inputCount': this.inputCount_
+        };
+    },
+
+    loadExtraState: function (state) {
+        this.inputCount_ = state['inputCount'];
+        this.updateShape_();
+    },
+
+    decompose: function (workspace) {
+        const containerBlock = workspace.newBlock('procedure_dependencies_mutator_container');
+        containerBlock.initSvg();
+        let connection = containerBlock.getInput('STACK').connection;
+        for (let i = 0; i < this.inputCount_; i++) {
+            const inputBlock = workspace.newBlock('procedure_dependencies_mutator_input');
+            inputBlock.nameValue_ = null;
+            inputBlock.initSvg();
+            connection.connect(inputBlock.previousConnection);
+            connection = inputBlock.nextConnection;
+        }
+        return containerBlock;
+    },
+
+    compose: function (containerBlock) {
+        let inputBlock = containerBlock.getInputTargetBlock('STACK');
+        const connections = [];
+        const fieldValues = {};
+        const fieldValuesDummy = [];
+        while (inputBlock && !inputBlock.isInsertionMarker()) {
+            connections.push(inputBlock.valueConnection_);
+            if (inputBlock.valueConnection_)
+                fieldValues[inputBlock.valueConnection_.sourceBlock_.id] = inputBlock.nameValue_;
+            else
+                fieldValuesDummy.push(inputBlock.nameValue_);
+            inputBlock = inputBlock.nextConnection && inputBlock.nextConnection.targetBlock();
+        }
+        for (let i = 0; i < this.inputCount_; i++) {
+            const connection = this.getInput('arg' + i) && this.getInput('arg' + i).connection.targetConnection;
+            if (connection && connections.indexOf(connection) === -1)
+                connection.disconnect();
+        }
+        this.inputCount_ = connections.length;
+        this.updateShape_();
+        const fieldValuesFlat = Object.values(fieldValues).concat(fieldValuesDummy);
+        let k = 0;
+        while (fieldValuesFlat.indexOf('dependency' + k) !== -1)
+            k++;
+        for (let i = 0, j = 0; i < this.inputCount_; i++) {
+            Blockly.Mutator.reconnect(connections[i], this, 'arg' + i);
+            const currentField = this.getField('name' + i);
+            const validator = currentField.getValidator();
+            currentField.setValidator(null);
+            if (connections[i])
+                currentField.setValue(fieldValues[connections[i].sourceBlock_.id] || '');
+            if (!connections[i] || currentField.getValue() === '')

```

Comment:
There is a lot going on here so it is only fair to expect some comments around this.

I am quite doubtful someone comes into this code and just understands this:

`(j < fieldValuesDummy.length && fieldValuesDummy[j++]) || 'dependency' + (k++)`

(amongst other locations where complex code in this file is not explained well)